programs/bpf_loader/src/lib.rs

```diff
@@ -535,8 +462,12 @@ fn process_instruction_common(
         if let Some(create_executor_metrics) = create_executor_metrics {
             create_executor_metrics.submit_datapoint(&mut invoke_context.timings);
         }
-
-        executor.execute(invoke_context)
+        match executor.program {
+            LoadedProgramType::Invalid => Err(InstructionError::IncorrectProgramId),
```

Comment:
I think for tomb stones we currently throw `InstructionError::InvalidAccountData` instead.